scripts/build_osm_network.py

```diff
@@ -0,0 +1,1051 @@
+
+
+def read_csv_nafix(file, **kwargs):
+    "Function to open a csv as pandas file and standardize the na value"
+    if "keep_default_na" not in kwargs:
+        kwargs["keep_default_na"] = False
+    if "na_values" not in kwargs:
+        kwargs["na_values"] = NA_VALUES
+
+    if os.stat(file).st_size > 0:
+        return pd.read_csv(file, **kwargs)
+    else:
+        return pd.DataFrame()
+
+
+def save_to_geojson(df, fn):
+    """
+    Save a (Geo)DataFrame to a GeoJSON file.
+
+    Parameters:
+    - df: The (Geo)DataFrame to be saved.
+    - fn: The filename (including the path) of the output GeoJSON file.
+
+    Returns:
+    None
+    """
+    if os.path.exists(fn):
+        os.unlink(fn)  # remove file if it exists
+
+    # save file if the (Geo)DataFrame is non-empty
+    if df.empty:
+        # create empty file to avoid issues with snakemake
+        with open(fn, "w") as fp:
+            pass
+    else:
+        # save file
+        df.to_file(fn, driver="GeoJSON")
+
+
+def read_geojson(fn, cols=[], dtype=None, crs="EPSG:4326"):
+    """
+    Function to read a geojson file fn. When the file is empty, then an empty
+    GeoDataFrame is returned having columns cols, the specified crs and the
+    columns specified by the dtype dictionary it not none.
+
+    Parameters:
+    ------------
+    fn : str
+        Path to the file to read
+    cols : list
+        List of columns of the GeoDataFrame
+    dtype : dict
+        Dictionary of the type of the object by column
+    crs : str
+        CRS of the GeoDataFrame
+    """
+    # if the file is non-zero, read the geodataframe and return it
+    if os.path.getsize(fn) > 0:
+        return gpd.read_file(fn)
+    else:
+        # else return an empty GeoDataFrame
+        df = gpd.GeoDataFrame(columns=cols, geometry=[], crs=crs)
+        if isinstance(dtype, dict):
+            for k, v in dtype.items():
+                df[k] = df[k].astype(v)
+        return df
+
+
+def to_csv_nafix(df, path, **kwargs):
+    """
+    Write a pandas DataFrame to a CSV file with NA values replaced.
+
+    Parameters:
+    - df: pandas DataFrame
+        The DataFrame to be written to the CSV file.
+    - path: str
+        The file path where the CSV file will be saved.
+    - **kwargs: keyword arguments
+        Additional arguments to be passed to the `to_csv` function of pandas.
+
+    Returns:
+    - None
+
+    If the DataFrame is not empty or does not have empty columns, it will be
+    written to the CSV file with NA values replaced by the first value in the
+    `NA_VALUES` list. If the DataFrame is empty or has empty columns, an empty
+    file will be created at the specified path.
+    """
+    if "na_rep" in kwargs:
+        del kwargs["na_rep"]
+    if not df.empty or not df.columns.empty:
+        return df.to_csv(path, **kwargs, na_rep=NA_VALUES[0])
+    else:
+        with open(path, "w") as fp:
+            pass
```

Comment:
I think calls to this function can be replaced simply by `df.to_csv(fn, quotechar="'")`. It is no problem to do this to an empty dataframe and the NaN values should be dealt with at reading.

Reply:
yes, done.